src/server/editorServices.ts

```diff
@@ -3515,7 +3523,7 @@ namespace ts.server {
             const typeAcquisition = proj.typeAcquisition!;
             Debug.assert(!!typeAcquisition, "proj.typeAcquisition should be set by now");
             // If type acquisition has been explicitly disabled, do not exclude anything from the project
-            if (typeAcquisition.enable === false) {
+            if (typeAcquisition.enable === false || typeAcquisition.inferTypings === false) {

```

Comment:
Just to clarify - does `applySafeList` only apply in the "infer from filenames" case?

Reply:
It removes any project files that appear in the safe list from the project, and adds them to the `include` list (for which typings are installed unconditionally).

If there is a case where we want to remove files from the include list _and_ the project, then this would be wrong (but I'm not sure what that is).